pkg/controller/scyllacluster/resource.go

```diff
@@ -219,9 +221,25 @@ func StatefulSetForRack(r scyllav1.RackSpec, c *scyllav1.ScyllaCluster, existing
 	if err != nil {
 		return nil, fmt.Errorf("can't get rack labels: %w", err)
 	}
-	matchLabels := helpers.ShallowCopyMap(rackLabels)
+	_, rackOrdinal, ok := slices.Find(c.Spec.Datacenter.Racks, func(rack scyllav1.RackSpec) bool {
+		return rack.Name == r.Name
+	})
+	if !ok {
+		return nil, fmt.Errorf("can't find ordinal of rack %q in ScyllaCluster %q", r.Name, naming.ObjRef(c))
+	}
+	rackLabels[naming.RackOrdinalLabel] = strconv.Itoa(rackOrdinal)
 	rackLabels[naming.ScyllaVersionLabel] = c.Spec.Version
 
+	selectorLabels, err := naming.RackLabels(r, c)
+	if err != nil {
+		return nil, fmt.Errorf("can't get selector labels: %w", err)
+	}
+
+	pvcLabels, err := naming.RackLabels(r, c)
```

Comment:
I suppose this could be more then just the selector labels, I guess the ordinal could be useful label as well. e.g. to list first node pvc in all clusters

Reply:
PVCTemplate is immutable